test/controller.test.ts

```diff
@@ -95,31 +107,80 @@ contract('Controller', ([owner, accountOwner1, accountOperator1, random]) => {
     })
   })
 
-  describe('Price', () => {
+  describe('Check if price is finalized', () => {
     before(async () => {
-      const batch = await controller.getBatch(otoken.address)
-      const expiryTimestampMock = new BigNumber('1598374220')
-      const priceMock = new BigNumber('200')
+      const lockingPeriod = new BigNumber(60) // 1min
       const disputePeriod = new BigNumber(60) // 1min
 
+      const batch = (await controller.getBatchDetails(otoken.address))[0]
+      // set batch oracle
       await oracle.setBatchOracle(batch, batchOracle.address)
+      // set locking and dispute period
+      await oracle.setLockingPeriod(batchOracle.address, lockingPeriod)
       await oracle.setDisputePeriod(batchOracle.address, disputePeriod)
+    })
+
+    it('should return false when price is not pushed to Oracle yet', async () => {
+      assert.equal(
+        await controller.isPriceFinalized(otoken.address),
+        false,
+        'Price is not finalized because it is not stored yet',
+      )
+    })
+
+    it('should return false when price is pushed and dispute period not over yet', async () => {
+      const expiryTimestampMock = new BigNumber(await time.latest())
+      const priceMock = new BigNumber('200')
+
+      const batch = (await controller.getBatchDetails(otoken.address))[0]
+      // increase time after locking period
+      await time.increase(61)
       await oracle.setBatchUnderlyingPrice(batch, expiryTimestampMock, priceMock)
+
+      const expectedResutl = await oracle.isDisputePeriodOver(batch, expiryTimestampMock)
+      assert.equal(
+        await controller.isPriceFinalized(otoken.address),
+        expectedResutl,
+        'Price is not finalized because dispute period is not over yet',
+      )
     })
 
-    it('should check if price is finalized or not', async () => {
-      const batch = await controller.getBatch(otoken.address)
-      const otokenExpiryTimestamp = new BigNumber('1598374220')
+    it('should return true when price is finalized', async () => {
+      const expiryTimestampMock = new BigNumber(await time.latest())
+      const priceMock = new BigNumber('200')
+
+      const batch = (await controller.getBatchDetails(otoken.address))[0]
+      // increase time after locking period
+      await time.increase(61)
+      await oracle.setBatchUnderlyingPrice(batch, expiryTimestampMock, priceMock)
+      // increase time after dispute period
+      await time.increase(60)
 
-      const expectedResutl = await oracle.isDisputePeriodOver(batch, otokenExpiryTimestamp)
-      assert.equal(await controller.isPriceFinalized(otoken.address), expectedResult, 'Price is not finalized')
+      const expectedResutl = await oracle.isDisputePeriodOver(batch, expiryTimestampMock)
+      assert.equal(await controller.isPriceFinalized(otoken.address), expectedResutl, 'Price is not finalized')
```

Comment:
can you also expectResult true or false? Is this testing for price is finalized or not finalized?

Reply:
This one is testing for price finalized:
- L155 price pushed
- L157 dispute period over

Reply:
@aparnakr updated `expectedResutl=true` so it doesn't make confusion.

Reply:
@aparnakr I updated all expected result to be a const input, idk why I assigned expected result from a call to the contract, def the wrong way to do it.